mod_reforged/hooks/skills/perks/perk_mastery_spear.nut

```diff
@@ -1,8 +1,126 @@
 ::Reforged.HooksMod.hook("scripts/skills/perks/perk_mastery_spear", function(q) {
-	q.onAfterUpdate <- function( _properties )
+	q.m.IsSpent <- true;
+	q.m.DamageTotalMult <- 0.75;
+
+	q.create = @(__original) function()
 	{
-		local spearwall = this.getContainer().getSkillByID("actives.spearwall");
-		if (spearwall != null && spearwall.m.ActionPointCost > 1)
-			spearwall.m.ActionPointCost -= 1;
+		__original();
+		this.m.Type = ::Const.SkillType.Perk | ::Const.SkillType.StatusEffect;
+		this.m.Description = ::Reforged.Mod.Tooltips.parseString("This character has mastered the art of fighting with the spear allowing him to attack for free once per [turn|Concept.Turn].");
+	}
+
+	q.isHidden = @() function()
+	{
+		return this.m.IsSpent;
+	}
+
+	q.getTooltip = @(__original) function()
+	{
+		local ret = __original();
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("The next spear attack costs " + ::MSU.Text.colorGreen("0") + " [Action Points|Concept.ActionPoints] and builds " + ::MSU.Text.colorGreen("0") + " [Fatigue|Concept.Fatigue] but does " + ::MSU.Text.colorizeMult(this.m.DamageMult) + " less damage")
+		});
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/warning.png",
+			text = "Will expire upon switching your weapon!"
+		});
+
+		return ret;
+	}
+
+	q.onAfterUpdate = @(__original) function( _properties )
+	{
+		__original(_properties);
+
+		foreach (skill in this.getContainer().m.Skills)
+		{
+			if (skill.getID() == "actives.spearwall")
+			{
+				skill.m.ActionPointCost -= 1;
+			}
+			else if (this.isSkillValid(skill))
+			{
+				skill.m.ActionPointCost = 0;
+				skill.m.FatigueCostMult *= 0;
+			}
+		}
+	}
+
+	q.onAffordablePreview = @(__original) function( _skill, _movementTile )
+	{
+		__original(_skill, _movementTile);
+
+		if (_skill != null)
+		{
+			local perk = this;
+			foreach (skill in this.getContainer().getSkillsByFunction((@(s) perk.isSkillValid(s))))
+			{
+				this.modifyPreviewField(skill, "ActionPointCost", 0, false);
+				this.modifyPreviewField(skill, "FatigueCostMult", 1, true);
```

Comment:
I understand that these MSU functions are confusing which is why I am moving to a new more intuitive affordability preview system in MSU. These functions here are correct though, what they basically say is that during preview this skill should modify the fields of other skills by this value INSTEAD of the modification that it does in its `onAfterUpdate` function. You could read up on the [MSU affordability preview documentation](https://github.com/MSUTeam/MSU/wiki/Affordability-Preview) for detail. But honesty, once the new more intuitive system is ready then it should be much more understandable.

I will also convert this to the new MSU system now.